receivers/envoyreceiver/envoy.go

```diff
@@ -85,8 +86,14 @@ func (r *envoyReceiver) Start(_ context.Context, host component.Host) error {
 }
 
 func (r *envoyReceiver) Shutdown(_ context.Context) error {
-	r.grpcServer.GracefulStop()
-	<-r.shutdownCh
+	if r.grpcServer != nil {
```

Comment:
Early return instead?

```suggestion
	if r.grpcServer == nil { 
	    r.logger.Warn("Shutting down envoy receiver that did not start successfully")
	    return
	}

	r.logger.Info("Shutting down envoy receiver gracefully")
	r.grpcServer.GracefulStop()
	<-r.shutdownCh
```